Lib/test/test_warnings/__init__.py

```diff
@@ -1186,7 +1211,7 @@ def __del__(self):
         rc, out, err = assert_python_ok("-c", code)
         # note: "__main__" filename is not correct, it should be the name
         # of the script
-        self.assertEqual(err, b'__main__:7: UserWarning: test')
+        self.assertEqual(err.decode(), '__main__:7: UserWarning: test')
```

Comment:
This may look unrelated, but comparing these as strings means a proper diff gets printed when the test fails (and it was this test that picked up the problem with the non-regex entry in the internal filter list).